src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/util/GroupUtil.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022 Exactpro (Exactpro Systems Limited)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.common.schema.message.impl.rabbitmq.group.util
+
+import com.exactpro.th2.common.grpc.AnyMessage
+import com.exactpro.th2.common.grpc.MessageGroup
+import com.exactpro.th2.common.grpc.MessageGroupBatch
+
+/**
+ * Transfers messages that match the [filterFunction] from current group to new group in [batchBuilder].
+ *
+ * If all messages are filtered no group will be added.
+ *
+ * If all messages match the filter the current group will be added into [batchBuilder] as is
+ */
+internal fun MessageGroup.filterMessagesToNewGroup(
+    batchBuilder: MessageGroupBatch.Builder,
+    filterFunction: (AnyMessage) -> Boolean,
+): List<AnyMessage> {
+    val matchCount = messagesList.count { filterFunction(it) }
```

Comment:
If filter-function is computation heavy invoking it twice to avoid new group creation might result in a negative performance yield. Not sure how lightweight our filters are

Reply:
Now filters are lightweight but they will be heavy after adding business filter and switching to Cbor body

Reply:
Looks fine for me